erc1155/src/lib.rs

```diff
@@ -0,0 +1,459 @@
+// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.4.2/contracts/token/ERC1155/IERC1155.sol
+
+#![no_std]
+#![feature(const_btree_new)]
+
+#[cfg(test)]
+use codec::{Decode, Encode};
+use gstd::{debug, exec, msg, prelude::*, ActorId};
+use primitive_types::U256;
+use scale_info::TypeInfo;
+
+pub mod base;
+
+const GAS_RESERVE: u64 = 500_000_000;
+const ZERO_ID: ActorId = ActorId::new([0u8; 32]);
+
+#[derive(Debug)]
+struct Erc1155Token {
+    name: String,
+    symbol: String,
+    description: String,
+    base_uri: String,
+    balances: BTreeMap<u128, BTreeMap<ActorId, u128>>,
```

Comment:
You must add the mapping from token ids to the owner

Reply:
fixed https://github.com/gear-tech/apps/commit/af0f12634f723d208f7de088ef4bacb910a139e0